src/App.tsx

```diff
@@ -1,24 +1,50 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+const maxItems = 42;
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [perPage, setPerPage] = useState<number>(5);
+  const toMax = perPage * currentPage;
+  const to = toMax < maxItems ? toMax : maxItems;
+  const from = (currentPage - 1) * perPage + 1;
+  const items = getNumbers(from, to).map(n => `Item ${n}`);
+
+  const setItemsPerPage = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setPerPage(+value);
+  };
+
+  const changePage = (page: number) => {
+    if (page === currentPage) {
+      return;
+    }
+
+    setCurrentPage(page);
+  };
+
+  useEffect(() => {
+    changePage(1);
+  }, [perPage]);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${from} - ${to} of ${maxItems})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
+            value={perPage}
+            onChange={(event) => setItemsPerPage(event)}
```

Comment:
```suggestion
            onChange={setItemsPerPage}
```
It will work the same